File: leetcode2325.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
      from string import ascii_lowercase
      s: str = ascii_lowercase
      d: dict = {}
      j: int = 0
      for i in key:
        if i not in d and i != ' ':
          d[i] = s[j]
          j += 1
      ans: str = ""
      for i in message:
        if i != ' ':
          ans += d[i]
        else:
          ans += ' '
      return ans
          
      

File: leetcode2325.java
class Solution {
    public String decodeMessage(String key, String message) {
      HashMap<Character,Character>map = new HashMap<>();
      String letters = "abcdefghijklmnopqrstuvwxyz";
      int j = 0;
      for(int i = 0;i<key.length(); i++){
        if(key.charAt(i) != ' '){
          if(!map.containsKey(key.charAt(i))){
            map.put(key.charAt(i),letters.charAt(j));
            j ++;
          }
        }
      }
      String result = "";
      for(int i = 0;i<message.length();i++){
        if(message.charAt(i) != ' '){
          result += map.get(message.charAt(i));
        }else{
          result += " ";
        }
      }
      return result;
    }
}
